sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -1340,6 +1340,15 @@ object GpuOverrides {
         override def convertToGpu(child: Expression): GpuExpression =
           GpuKnownFloatingPointNormalized(child)
       }),
+    expr[KnownNotNull](
+      "Tag an expression as known to not be null",
+      ExprChecks.unaryProjectNotLambdaInputMatchesOutput(
+        TypeSig.commonCudfTypes + TypeSig.DECIMAL,
+        TypeSig.commonCudfTypes + TypeSig.DECIMAL),
```

Comment:
Technically the second param should be `TypeSig.all` because Spark supports all types with this, although in reality it is only ever going to use it on byte, short, int, long, float, and double.  All of the other values are not primitive types and as such can be nullable in a UDF.

Reply:
Updated, however does it mean the `KnownNotNull` also supports `NullType`, which I thought should not be allowed for this expression?